pkg/controller/certificatesigningrequests/sync.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificatesigningrequests
+
+import (
+	"context"
+	"fmt"
+
+	certificatesv1 "k8s.io/api/certificates/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager"
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/jetstack/cert-manager/pkg/controller/certificatesigningrequests/util"
+	logf "github.com/jetstack/cert-manager/pkg/logs"
+)
+
+func (c *Controller) Sync(ctx context.Context, csr *certificatesv1.CertificateSigningRequest) (err error) {
```

Comment:
nit `err error` (naming the return parameter) makes writing methods confusing, because `return` on its own is then a valid thing to do (and the contents of `err` gets.. murky depending on the code before it). Is there a particular reason we use a named return parameter here?

Reply:
Thanks, I agree. No reason was just a copy paste job.